src/vmdssd.c

```diff
@@ -71,24 +72,34 @@ static char *get_slot_from_syspath(char *path)
 	return ret;
 }
 
-static int check_slot_module(const char *slot_path)
+bool vmdssd_check_slot_module(const char *slot_path)
 {
-	char module_path[PATH_MAX], real_module_path[PATH_MAX];
-	struct list dir;
-
-	// check if slot is managed by pciehp driver
-	snprintf(module_path, PATH_MAX, "%s/module", slot_path);
-	if (scan_dir(module_path, &dir) == 0) {
-		list_erase(&dir);
-		if (realpath(module_path, real_module_path) == NULL)
-			return -1;
-		if (strcmp(real_module_path, SYSFS_PCIEHP) != 0)
-			__set_errno_and_return(EINVAL);
-	} else {
-		__set_errno_and_return(ENOENT);
+	char domain_path[PATH_MAX], real_domain_path[PATH_MAX];
+	char *address, *domain;
+	struct cntrl_device *cntrl;
+
+	// check if slot address contains vmd domain
+	list_for_each(sysfs_get_cntrl_devices(), cntrl) {
+		if (cntrl->cntrl_type == CNTRL_TYPE_VMD) {
+			snprintf(domain_path, PATH_MAX, "%s/%s/domain",
+				 SYSFS_VMD, basename(cntrl->sysfs_path));
```

Comment:
as discussed, the best will be keep information with controller, to avoid repeated realpath calls